javascript/computer_science/time_complexity.md

```diff
@@ -240,27 +240,27 @@ function findValue(arr) {
 
 In the worst case (Big O), which would happen if the item is not in the array, we would say it had linear complexity `O(N)`. This is because the `item` we are looking for is not in the array, so our code must iterate on every value. If the array input doubles in size then the worst case also means our method must double the number of iterations looking for the `item`.
 
-However in the best case scenario the value we are looking for will be the first item in the array. In this case our algorithm takes just one step. This has a complexity of `O(1)`. This is its Omega Complexity
+However, in the best-case scenario the value we are looking for will be the first item in the array. In this case our algorithm takes just one step. This has a complexity of `O(1)`. This is its Omega Complexity.
 
 Omega Notation isn't considered as useful because it is unlikely our item will often be the first item in our data structure search, so it doesn't give us any idea how well the algorithm will scale.
 
 #### Big-Θ (Big-Theta Notation)
 
-While Omega Notation measures the best case scenario for an algorithm's efficiency, and Big O measures the worst case, Theta looks to give the exact value or a useful range between narrow upper and lower bounds.
+While Omega Notation measures the best-case scenario for an algorithm's efficiency, and Big O measures the worst case, Theta looks to give the exact value or a useful range between narrow upper and lower bounds.
 
-If we had some code that looped every item in an array then it doesn't matter the size of the array, our algorithm will always run in `O(N)` time in it's best and worst case scenario. In that case we know it's exact performance in all scenarios is `O(N)` and that is the Theta performance of our algorithm. For other algorithms then Theta may represent both the lower and upper bound of an algorithm that has different complexities. We won't get into this more here because Big O is the primary notation used for general algorithm time complexity.
+If we had some code that looped every item in an array, then it doesn't matter the size of the array. Our algorithm will always run in `O(N)` time in its best- and worst-case scenario. In that case we know it's exact performance in all scenarios is `O(N)`, and that is the Theta performance of our algorithm. For other algorithms then Theta may represent both the lower and upper bound of an algorithm that has different complexities. We won't get into this more here because Big O is the primary notation used for general algorithm time complexity.
```

Comment:
Overall this looks good. However, the dash after the best isn't connecting to a word and seems confusing to me. If we could remove the dash from the "best-" or change it to "best-case" I'll go ahead and approve it.

Reply:
That's fine. Suspended hyphens are grammatically correct, but I agree it could be confusing. We can change it to "best-case" then if that's ok. Also, reading it again, I think "scenario" should be "scenarios".  I'm new to doing pull requests, but I'll see if I can edit it again and submit the new changes! Thanks!